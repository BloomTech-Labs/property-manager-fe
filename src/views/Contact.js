/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable react/prop-types */

import React, { Component } from 'react';

// icons import
import emailIcon from '../assets/img/WhiteEmail.png';
import phoneIcon from '../assets/img/WhitePhone.png';
import facebookIcon from '../assets/img/WhiteFacebook.png';
import twitterIcon from '../assets/img/WhiteTwitter.png';

// ---------- Styles ---------- //

const encode = data => {
  return (
    Object.keys(data)
      // eslint-disable-next-line prefer-template
      .map(key => encodeURIComponent(key) + '=' + encodeURIComponent(data[key]))
      .join('&')
  );
};

export default class Contact extends Component {
  constructor(props) {
    super(props);
    this.state = { name: '', email: '', message: '' };
  }

  handleSubmit = e => {
    e.preventDefault();
    fetch('/', {
      method: 'POST',
      headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
      body: encode({ 'form-name': 'contact', ...this.state })
    });
  };

  handleChange = e => this.setState({ [e.target.name]: e.target.value });

  render() {
    const { name, email, message } = this.state;
    return (
      <div className="contactPage">
        <form className="cm-form" onSubmit={this.handleSubmit}>
          <h1>Contact Me</h1>
          <div className="form-elements">
            <label>
              <div>Your Name: </div>
              <input
                type="text"
                name="name"
                value={name}
                onChange={this.handleChange}
              />
            </label>
            <label>
              <div>Your Email: </div>
              <input
                type="email"
                name="email"
                value={email}
                onChange={this.handleChange}
              />
            </label>
            <label>
              <div>Message: </div>
              <textarea
                name="message"
                value={message}
                rows="4"
                onChange={this.handleChange}
              />
            </label>
          </div>
          <div className="form-btns">
            <button type="button">Send</button>
          </div>
        </form>
        <div className="contactInfo">
          <h3>ACME, THE HI-FI PROTOTYPING COMPANY</h3>
          <div className="contactType">
            <img src={emailIcon} alt="email" width="25px" />
            <h3>acme.info@2mail.com</h3>
          </div>
          <div className="contactType">
            <img src={phoneIcon} alt="phone" width="25px" />
            <h3>+0 000-000-0000</h3>
          </div>
          <h3 className="followUs">FOLLOW US</h3>
          <div className="contactIcons">
            <img src={facebookIcon} alt="fb" width="35px" />
            <img src={twitterIcon} alt="twitter" width="35px" />
          </div>
        </div>
      </div>
    );
  }
}