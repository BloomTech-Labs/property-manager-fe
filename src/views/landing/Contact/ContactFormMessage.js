/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/prop-types */
import React from 'react';
import { navigate } from '@reach/router';
import {
  Button,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogActions,
  Container
} from '@material-ui/core';
import MailSent from '../../../assets/svg/mail-sent.svg';

function ContactFormMessage({ open, handleClose, Transition }) {
  return (
    <>
      <Dialog
        open={open}
        TransitionComponent={Transition}
        keepMounted
        onClose={handleClose}
        aria-labelledby="alert-dialog-slide-title"
        aria-describedby="alert-dialog-slide-description"
      >
        <Container>
          <DialogTitle disableTypography id="alert-dialog-slide-title">
            <h5 className="contactModalHeader">Form Submitted!</h5>
          </DialogTitle>
          <img
            className="contactFormImg"
            src={MailSent}
            alt="Thank you for message"
          />
          <DialogContent>
            <p>Thank you for your feedback!</p>
          </DialogContent>
          <DialogActions>
            <Button autoFocus color="primary" onClick={() => navigate('/')}>
              Close
            </Button>
          </DialogActions>
        </Container>
      </Dialog>
    </>
  );
}

export default ContactFormMessage;
