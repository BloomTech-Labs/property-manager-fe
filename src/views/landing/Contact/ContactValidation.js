/* eslint-disable import/prefer-default-export */
import * as Yup from 'yup';

export const contactValidation = Yup.object().shape({
  name: Yup.string()
    .min(2)
    .max(100)
    .required('Must enter a Name'),
  email: Yup.string()
    .email('Must be a valid email address')
    .max(255, 'Email entered was too long')
    .required('Must enter an Email Address'),
  message: Yup.string()
    .min(8, 'Message must be at least 8 characters')
    .max(255, 'Message is too long')
    .required('Must enter a Message')
});
