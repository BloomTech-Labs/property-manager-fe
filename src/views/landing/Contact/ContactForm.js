/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { Formik, Field, Form } from 'formik';
import { Slide } from '@material-ui/core';
import { MdEmail, MdMessage, MdPerson } from 'react-icons/md';
import FormErrors from '../../../helpers/FormErrors';
import { I } from '../../../components/UI';
import ContactFormMessage from './ContactFormMessage';
import { contactValidation } from './ContactValidation';

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

export default function ContactForm() {
  // modal logic
  const [open, setOpen] = React.useState(false);

  const handleOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  // form logic
  const encode = data => {
    return Object.keys(data)
      .map(
        // eslint-disable-next-line prefer-template
        key => encodeURIComponent(key) + '=' + encodeURIComponent(data[key])
      )
      .join('&');
  };

  return (
    <div className="contact-form-card">
      <h2 className="desktop-form">Contact Us</h2>
      <Formik
        validationSchema={contactValidation}
        initialValues={{
          name: '',
          email: '',
          message: '',
          'bot-field': '',
          'form-name': 'contact'
        }}
        onSubmit={(values, { resetForm }) => {
          fetch('/', {
            method: 'POST',
            headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
            body: encode({
              'form-name': 'contact',
              ...values
            })
          })
            .then(() => {
              resetForm();
              handleOpen();
            })
            // eslint-disable-next-line no-console
            .catch(err => console.error(err));
        }}
      >
        {({ errors, touched, isSubmitting }) => (
          <Form data-testid="form-element">
            <Field type="hidden" name="bot-field" />
            <Field type="hidden" name="contact" />
            <div className="input-wrapper">
              <label htmlFor="name">
                <I>
                  <MdPerson />
                </I>{' '}
                Name
              </label>
              <Field placeholder="Enter your name *" name="name" type="text" />
              <FormErrors touched={touched.name} message={errors.name} />
            </div>
            <div className="input-wrapper">
              <label htmlFor="email">
                <I>
                  <MdEmail />
                </I>{' '}
                Email
              </label>
              <Field
                placeholder="Enter your email address *"
                name="email"
                type="email"
              />
              <FormErrors touched={touched.email} message={errors.email} />
            </div>
            <div className="input-wrapper">
              <label htmlFor="message">
                <I>
                  <MdMessage />
                </I>{' '}
                Message
              </label>
              <Field
                as="textarea"
                rows="4"
                placeholder="Enter your message *"
                name="message"
                type="text"
              />
              <FormErrors touched={touched.message} message={errors.message} />
            </div>
            <div className="submit-btn-wrapper">
              <button
                data-testid="contactBtn"
                className="btn button"
                type="submit"
                disabled={isSubmitting}
              >
                Submit
              </button>
            </div>
          </Form>
        )}
      </Formik>
      <ContactFormMessage
        open={open}
        handleClose={handleClose}
        Transition={Transition}
      />
    </div>
  );
}
