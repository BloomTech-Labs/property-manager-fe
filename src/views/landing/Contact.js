/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable react/prop-types */

import React from 'react';

// contact form
// icons import
import { FaFacebook, FaTwitter } from 'react-icons/fa';
import { Container } from '@material-ui/core';
import ContactForm from '../../components/ContactForm/ContactForm';

export default function Contact() {
  return (
    <div className="contact-page">
      <Container>
        <div className="contact-content">
          <ContactForm />
          <div className="contactInfo">
            <h2>PropMan</h2>
            <div className="contactType">
              <p>
                Feel free to contact us if you have any questions about PropMan
                and how it can improve your life as a landlord or tenant.
              </p>
            </div>
            <h3>Follow Us</h3>
            <div className="contactIcons">
              <FaFacebook className="icon" />
              <FaTwitter className="icon" />
            </div>
          </div>
        </div>
      </Container>
    </div>
  );
}
