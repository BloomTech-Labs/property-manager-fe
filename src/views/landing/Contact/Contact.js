import React from 'react';

// contact form
// icons import
import { FaInstagram, FaTwitter, FaFacebookSquare } from 'react-icons/fa';
import { Container } from '@material-ui/core';
import ContactForm from './ContactForm';

export default function Contact() {
  return (
    <div className="contact-page">
      <Container>
        <div className="contact-content">
          <ContactForm />
          <div className="contactInfo">
            <h2 className="desktop-contact">Freehold</h2>
            <div className="contactType desktop-contact">
              <p>
                Feel free to contact us if you have any questions about Freehold
                and how it can improve your life as a landlord or tenant.
              </p>
            </div>
            <h3 className="desktop-contact">Follow Us</h3>
            <div className="contactIcons">
              <a
                className="icon-link"
                target="_blank"
                rel="noopener noreferrer"
                href="https://www.instagram.com/_freehold_/"
              >
                <FaInstagram className="icon" />
              </a>
              <a
                className="icon-link"
                target="_blank"
                rel="noopener noreferrer"
                href="https://twitter.com/Freehol91308053"
              >
                <FaTwitter className="icon" />
              </a>
              <a
                className="icon-link"
                target="_blank"
                rel="noopener noreferrer"
                href="https://www.facebook.com/freehold.devs.9"
              >
                <FaFacebookSquare className="icon" />
              </a>
            </div>
          </div>
        </div>
      </Container>
    </div>
  );
}
