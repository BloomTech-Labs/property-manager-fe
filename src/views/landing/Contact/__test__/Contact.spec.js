import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Contact from '../Contact';

afterEach(rtl.cleanup);

test('Checks the document for the word Freehold', async () => {
  const wrapper = rtl.render(<Contact />);
  expect(await wrapper.queryAllByText(/freehold/i));
});

test('snapshot', () => {
  const { container } = rtl.render(<Contact />);
  expect(container.firstChild).toMatchSnapshot();
});
