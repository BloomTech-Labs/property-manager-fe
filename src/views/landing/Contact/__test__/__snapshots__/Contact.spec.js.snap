// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="contact-page"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="contact-content"
    >
      <div
        class="contact-form-card"
      >
        <h2
          class="desktop-form"
        >
          Contact Us
        </h2>
        <form
          action="#"
          data-testid="form-element"
        >
          <input
            name="bot-field"
            type="hidden"
            value=""
          />
          <input
            name="contact"
            type="hidden"
            value=""
          />
          <div
            class="input-wrapper"
          >
            <label
              for="name"
            >
              <span
                class="css-1bcyk0w"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </span>
               
              Name
            </label>
            <input
              name="name"
              placeholder="Enter your name *"
              type="text"
              value=""
            />
          </div>
          <div
            class="input-wrapper"
          >
            <label
              for="email"
            >
              <span
                class="css-1bcyk0w"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
              </span>
               
              Email
            </label>
            <input
              name="email"
              placeholder="Enter your email address *"
              type="email"
              value=""
            />
          </div>
          <div
            class="input-wrapper"
          >
            <label
              for="message"
            >
              <span
                class="css-1bcyk0w"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-2 12H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
                  />
                </svg>
              </span>
               
              Message
            </label>
            <textarea
              name="message"
              placeholder="Enter your message *"
              rows="4"
              type="text"
            />
          </div>
          <div
            class="submit-btn-wrapper"
          >
            <button
              class="btn button"
              data-testid="contactBtn"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
      <div
        class="contactInfo"
      >
        <h2
          class="desktop-contact"
        >
          Freehold
        </h2>
        <div
          class="contactType desktop-contact"
        >
          <p>
            Feel free to contact us if you have any questions about Freehold and how it can improve your life as a landlord or tenant.
          </p>
        </div>
        <h3
          class="desktop-contact"
        >
          Follow Us
        </h3>
        <div
          class="contactIcons"
        >
          <a
            class="icon-link"
            href="https://www.instagram.com/_freehold_/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              />
            </svg>
          </a>
          <a
            class="icon-link"
            href="https://twitter.com/Freehol91308053"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              />
            </svg>
          </a>
          <a
            class="icon-link"
            href="https://www.facebook.com/freehold.devs.9"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 32H48A48 48 0 0 0 0 80v352a48 48 0 0 0 48 48h137.25V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.27c-30.81 0-40.42 19.12-40.42 38.73V256h68.78l-11 71.69h-57.78V480H400a48 48 0 0 0 48-48V80a48 48 0 0 0-48-48z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
